workflows/taxprofiler.nf

```diff
@@ -113,65 +113,63 @@ workflow TAXPROFILER {
     // Validate input files and create separate channels for FASTQ, FASTA, and Nanopore data
     samplesheet
         .branch { meta, run_accession, instrument_platform, fastq_1, fastq_2, fasta ->
-            //println "Mapping: meta=$meta, run_accession=$run_accession, instrument_platform=$instrument_platform, fastq_1=$fastq_1, fastq_2=$fastq_2, fasta=$fasta"
+            meta.run_accession = run_accession
+            meta.instrument_platform = instrument_platform
 
-        meta.run_accession = run_accession
-        meta.instrument_platform = instrument_platform
+            // Define single_end based on the conditions
+            meta.single_end = ( fastq_1 && !fastq_2 && instrument_platform != 'OXFORD_NANOPORE' )
 
-        // Define single_end based on the conditions
-        meta.single_end = (fastq_1 && !fastq_2 && instrument_platform != 'OXFORD_NANOPORE')
+            // Define is_fasta based on the presence of fasta
+            meta.is_fasta = fasta ? true : false
 
-        // Define is_fasta based on the presence of fasta
-        meta.is_fasta = fasta ? true : false
-
-        if (!meta.is_fasta && !fastq_1) {
-            error("ERROR: Please check input samplesheet: entry `fastq_1` doesn't exist!")
-        }
-        if (meta.instrument_platform == 'OXFORD_NANOPORE' && fastq_2) {
-            error("Error: Please check input samplesheet: for Oxford Nanopore reads entry `fastq_2` should be empty!")
-        }
-        if (meta.single_end && fastq_2) {
-            error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
+            if ( !meta.is_fasta && !fastq_1 ) {
+                error("ERROR: Please check input samplesheet: entry `fastq_1` doesn't exist!")
+            }
+            if ( meta.instrument_platform == 'OXFORD_NANOPORE' && fastq_2 ) {
+                error("Error: Please check input samplesheet: for Oxford Nanopore reads entry `fastq_2` should be empty!")
+            }
+            if ( meta.single_end && fastq_2 ) {
+                error("Error: Please check input samplesheet: for single-end reads entry `fastq_2` should be empty")
```

Comment:
I would rather not do validation within the `branch` itself, better to have a separate map function for doing this.

Reply:
I will re-organise this section.